src/libraries/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -11,6 +11,19 @@ public static partial class Enumerable
     {
         public static IEnumerable<TSource> AsEnumerable<TSource>(this IEnumerable<TSource> source) => source;
 
+        /// <summary>Returns an empty <see cref="IEnumerable{TResult}"/>.</summary>
+        public static IEnumerable<TResult> Empty<TResult>() =>
+            Array.Empty<TResult>(); // explicitly not using [] in case the compiler ever changed to using Enumerable.Empty
+
+        /// <summary>Gets whether the enumerable is empty and is known to always be empty.</summary>
+        /// <remarks>
+        /// This can only return true for empty things that are known to then always be empty, like an array.
+        /// It must not return true for things that are currently empty but could become non-empty, like a list.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsImmutableEmpty<TSource>(IEnumerable<TSource> source) =>
```

Comment:
The method name and preceding comments seem to suggest a predicate that covers a broader range of types than what the implementation is doing. Shouldn't this just be `IsEmptyArray`?

Reply:
I wanted to accommodate other things in the future without a rename (this started out handling both empty arrays and the empty partition type). But I can rename it.

Reply:
Do we care enough about empty immutable arrays?

Reply:
> Do we care enough about empty immutable arrays?

Probably not.